autoload/lsp/utils/range.vim

```diff
@@ -70,3 +70,23 @@ function! lsp#utils#range#lsp_to_vim(bufnr, range) abort
 
     return l:position
 endfunction
+
+function! lsp#utils#range#is_valid_for_buffer(bufnr, range)
+    let [l:start_line, l:start_col] = lsp#utils#position#lsp_to_vim(a:bufnr, a:range['start'])
+    let [l:end_line, l:end_col] = lsp#utils#position#lsp_to_vim(a:bufnr, a:range['end'])
+
+    let l:buf_linecount = getbufinfo(a:bufnr)[0]['linecount']
+
+    if l:buf_linecount < l:start_line || l:buf_linecount < l:end_line
+        return v:false
+    endif
+
+    let l:first_line_length = len(getbufline(a:bufnr, l:start_line, l:end_line)[0])
+    let l:last_line_length = len(getbufline(a:bufnr, l:start_line, l:end_line)[-1])
+
+    if l:first_line_length < l:start_col || l:last_line_length < l:end_col
+        return v:false
+    endif
+
+    return v:true
+endfunction
```

Comment:
I think we should add `range#_get_buffer_range` and `range#_contains`.

The `range#_contains` implemented in here (this PR stale now due to nvim_extmark issues)
https://github.com/prabirshrestha/vim-lsp/pull/1022/files#diff-9f982ec39267f1f987caa856add5947c949fb085afd6c4679f47917f21ed88daR32